app/helpers/admin/site_maps_helper.rb

```diff
@@ -7,7 +7,7 @@ module Admin::SiteMapsHelper
   #
   # @!visibility private
   def sitemap_pages
-    @sitemap_pages ||= proper_pages.only(:id, :name, :page_id, :path).group_by(&:page_id)
+    @sitemap_pages ||= proper_pages.only(:id, :name, :page_id, :path, :position, :role).group_by(&:page_id)
```

Comment:
As per the other `only` scope, this fields are needed to render the site map. If they're not specified here then it's not possible to order pages or draw a special icon depending on the role.